map/bookmark_manager.cpp

```diff
@@ -1549,6 +1551,18 @@ std::string BookmarkManager::GetCategoryFileName(kml::MarkGroupId categoryId) co
   return GetBmCategory(categoryId)->GetFileName();
 }
 
+kml::MarkGroupId BookmarkManager::GetCategoryByFileName(std::string const & fileName) const
```

Comment:
1. Is it possible that the same file will change its name after some iCloud/other cloud updates?
2. If the same file is imported, it will always be duplicated now, right? Any ideas on how to handle it?

Reply:
1. No. Other file == (Old was deleted + New one created).
2. During the importing the New name will be generated and passed to the 
`void BookmarkManager::CreateCategories`. See:
https://github.com/organicmaps/organicmaps/blob/621eaaf67a0b44f415d2389e02da58a99eb5be72/map/bookmark_manager.cpp#L2002-L2022

For the cloud sync, there will be a new method `ReloadBookmarkRoutine` that skips new name generation and only re-load the data preserving the fileName.